From 1eb0626c0e2b1de2ba29d5418113707266346505 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:47:49 -0400
Subject: [drm] Use list_for_each_entry() for bus traversal

Message-id: <20140806034749.1438.40315.stgit@gir.stowe>
Patchwork-id: 87444
O-Subject: [RHEL7.1 PATCH 35/69] drm: Use list_for_each_entry() for bus traversal
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 59c1ad3b82b96bfcb40bf96ecff1d94f218fca6e

commit 59c1ad3b82b96bfcb40bf96ecff1d94f218fca6e
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Feb 13 21:14:00 2014 +0800

    drm: Use list_for_each_entry() for bus traversal

    Replace list_for_each() + pci_bus_b() with list_for_each_entry().

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/drm_fops.c b/drivers/gpu/drm/drm_fops.c
index 3f84277..2e0cd72 100644
--- a/drivers/gpu/drm/drm_fops.c
+++ b/drivers/gpu/drm/drm_fops.c
@@ -325,7 +325,8 @@ static int drm_open_helper(struct inode *inode, struct file *filp,
    pci_dev_put(pci_dev);
   }
   if (!dev->hose) {
-   struct pci_bus *b = pci_bus_b(pci_root_buses.next);
+   struct pci_bus *b = list_entry(pci_root_buses.next,
+    struct pci_bus, node);
    if (b)
     dev->hose = b->sysdata;
   }
-- 
1.7.1